runtime/runtime.go

```diff
@@ -0,0 +1,58 @@
+package runtime
+
+import (
+	"context"
+	"log"
+	"time"
+
+	"github.com/srl-labs/containerlab/runtime/docker"
+	"github.com/srl-labs/containerlab/types"
+)
+
+const (
+	DockerRuntime     = "docker"
+	ContainerdRuntime = "containerd"
+)
+
+type ContainerRuntime interface {
+	// Set the network management details (generated by the config.go)
+	SetMgmtNet(types.MgmtNet)
+	// Create container (bridge) network
+	CreateNet(context.Context) error
+	// Delete container (bridge) network
+	DeleteNet(context.Context) error
+	// Pull container image if not present
+	PullImageIfRequired(context.Context, string) error
+	// Create container
+	CreateContainer(context.Context, *types.Node) error
+	// Start pre-created container by its name
+	StartContainer(context.Context, string) error
+	// Stop running container by its name
+	StopContainer(context.Context, string, *time.Duration) error
+	// Create a creates a symlink for containers network namespace so that it can be managed by iproute2 utility
+	LinkContainerNS(string, string) error
```

Comment:
wdyt about removing this from runtime signature? this looks to me more like an utils function that just acts on a container name, but the runtime details are not in the scope of this function

Reply:
yep, makes sense